test/c/test_sabre_layout.c

```diff
@@ -0,0 +1,325 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+#include "common.h"
+#include <complex.h>
+#include <math.h>
+#include <qiskit.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+void print_circuit(const QkCircuit *qc) {
+    size_t num_instructions = qk_circuit_num_instructions(qc);
+    QkCircuitInstruction inst;
+    for (size_t i = 0; i < num_instructions; i++) {
+        qk_circuit_get_instruction(qc, i, &inst);
+        printf("%s: qubits: (", inst.name);
+        for (uint32_t j = 0; j < inst.num_qubits; j++) {
+            printf("%d,", inst.qubits[j]);
+        }
+        printf(")");
+        uint32_t num_clbits = inst.num_clbits;
+        if (num_clbits > 0) {
+            printf(", clbits: (");
+            for (uint32_t j = 0; j < num_clbits; j++) {
+                printf("%d,", inst.clbits[j]);
+            }
+            printf(")");
+        }
+        printf("\n");
+        qk_circuit_instruction_clear(&inst);
+    }
+}
+
+bool compare_circuits(const QkCircuit *res, const QkCircuit *expected) {
+    if (qk_circuit_num_instructions(res) != qk_circuit_num_instructions(expected)) {
+        printf("Number of instructions in circuit is mismatched");
+        return false;
+    }
+    QkCircuitInstruction *res_inst = malloc(sizeof(QkCircuitInstruction));
+    QkCircuitInstruction *expected_inst = malloc(sizeof(QkCircuitInstruction));
```

Comment:
why `malloc` and not stack vars?

Reply:
I assume I copied this from somewhere. I think I had circuit tests I wrote like this mostly to test `sizeof()`. I'll change it to use stack vars instead.